templates/data-providers/index.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { TextField, Button, Card } from '@oacore/design'
+
+import styles from './styles.module.css'
+import SuccessMessage from './success-message'
+
+const DataProviderPageTemplate = React.forwardRef(
+  (
+    {
+      url,
+      onSubmit,
+      onUrlChange,
+      helperMessage,
+      variant,
+      showSuccessMessage,
+      oaiPmhUrl,
+      isFormValid = false,
+    },
+    ref
+  ) => {
+    const handleSubmit = (event) => {
+      event.preventDefault()
+      if (onSubmit) onSubmit(event)
```

Comment:
Maybe this should be like
```suggestion
      if (onSubmit) onSubmit(event)
      if (!event.defaultPrevented) event.preventDefault()
```

What do you think?

Reply:
I guess it's the same and what's the point of checking if the event was prevented by default. Doesn't it change something?

Reply:
It's not the same. You allow the child component to prevent. Basically, you can have different logic if the event was prevented. My example was stupid.

I added a condition because I did not check the docs. The method might throw if it was already prevented.